ui/StatusQ/src/statussyntaxhighlighter.cpp

```diff
@@ -101,3 +145,89 @@ void StatusSyntaxHighlighter::setCodeForegroundColor(const QColor& color)
     m_codeForegroundColor = color;
     emit codeForegroundColorChanged();
 }
+
+QColor StatusSyntaxHighlighter::hyperlinkColor() const
+{
+    return m_hyperlinkColor;
+}
+
+void StatusSyntaxHighlighter::setHyperlinkColor(const QColor& color)
+{
+    if(color == m_hyperlinkColor) return;
+    m_hyperlinkColor = color;
+    emit hyperlinkColorChanged();
+}
+
+QColor StatusSyntaxHighlighter::hyperlinkHoverColor() const
+{
+    return m_hyperlinkHoverColor;
+}
+
+void StatusSyntaxHighlighter::setHyperlinkHoverColor(const QColor& color)
+{
+    if(color == m_hyperlinkHoverColor) return;
+    m_hyperlinkHoverColor = color;
+    emit hyperlinkHoverColorChanged();
+}
+
+QStringList StatusSyntaxHighlighter::hyperlinks() const
+{
+    return m_hyperlinks;
+}
+
+void StatusSyntaxHighlighter::setHyperlinks(const QStringList& hyperlinks)
+{
+    if(hyperlinks == m_hyperlinks) return;
+    m_hyperlinks = hyperlinks;
+    emit hyperlinksChanged();
+}
+QString StatusSyntaxHighlighter::highlightedHyperlink() const
+{
+    return m_highlightedHyperlink;
+}
+
+void StatusSyntaxHighlighter::setHighlightedHyperlink(const QString& hyperlink)
+{
+    if(hyperlink == m_highlightedHyperlink) return;
+    m_highlightedHyperlink = hyperlink;
+    emit highlightedHyperlinkChanged();
+}
+
+QRegularExpression StatusSyntaxHighlighter::highlightedHyperlinkRegularExpression() const
+{
+    const auto possibleUrlFormats = getPossibleUrlFormats(QUrl(m_highlightedHyperlink));
+    return buildHyperlinkRegex(possibleUrlFormats);
+}
+
+QRegularExpression StatusSyntaxHighlighter::hyperlinksRegularExpression() const
+{
+    QStringList hyperlinks;
+    for(const QString& hyperlink : qAsConst(m_hyperlinks))
+    {
+        const auto possibleUrlFormats = getPossibleUrlFormats(QUrl(hyperlink));
+        hyperlinks.append(possibleUrlFormats);
+    }
+
+    return buildHyperlinkRegex(hyperlinks);
+}
+
+QStringList StatusSyntaxHighlighter::getPossibleUrlFormats(const QUrl& url) const
+{
+    QStringList result;
+    result.append(QRegularExpression::escape(url.toString()));
+    result.append(QRegularExpression::escape(url.toString(QUrl::EncodeUnicode)));
+    result.append(QRegularExpression::escape(url.toString(QUrl::FullyEncoded)));
```

Comment:
I think then this wouldn't be needed; `QUrl` has an operator `==`: https://doc.qt.io/qt-5/qurl.html#operator-eq-eq

Reply:
Or even this could be of help: https://doc.qt.io/qt-5/qurl.html#matches